### PR TITLE
ENG-2237 user should only see what groups he/she has access to

### PR DESCRIPTION
**Pull request by jeffgo10**
note: pls review thoroughly as this has a lot of user group logic removed around group redux codebase

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **jeffgo10**

If they are not already assigned, you can assign the PR to them by writing `/assign @jeffgo10` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/ui/pages/config/SinglePageSettingsFormContainer.js
Comment (ichalagashvili) [inline src/ui/pages/config/SinglePageSettingsFormContainer.js:25]: dont we want all groups fetched here without pagination?

File: src/ui/users/my-profile/AppSettingsFormContainer.js
Comment (ichalagashvili) [inline src/ui/users/my-profile/AppSettingsFormContainer.js:26]: same here

File: src/ui/widgets/clone/CloneWidgetFormContainer.js
Comment (ichalagashvili) [inline src/ui/widgets/clone/CloneWidgetFormContainer.js:57]: and here

File: src/ui/widgets/common/WidgetFormContainer.js
Comment (ichalagashvili) [inline src/ui/widgets/common/WidgetFormContainer.js:39]: and here

File: src/ui/pages/config/SinglePageSettingsFormContainer.js
Comment (ichalagashvili) [reply src/ui/pages/config/SinglePageSettingsFormContainer.js:25]: ah I see. Maybe it makes sense to rename `fetchGroups` to `fetchMyGroups` to be clearer?

File: src/ui/widgets/clone/CloneWidgetFormContainer.js
Comment (ichalagashvili) [reply src/ui/widgets/clone/CloneWidgetFormContainer.js:57]: here to we want `fetchGroups` or `fetchAllGroupEntries` ? or maybe both?

File: src/ui/widgets/common/WidgetFormContainer.js
Comment (ichalagashvili) [reply src/ui/widgets/common/WidgetFormContainer.js:39]: here to we want `fetchGroups` or `fetchAllGroupEntries` ? or maybe both?

File: src/ui/widgets/edit/EditWidgetFormContainer.js
Comment (ichalagashvili) [inline src/ui/widgets/edit/EditWidgetFormContainer.js:34]: here to we want `fetchGroups` or `fetchAllGroupEntries` ? or maybe both?

File: src/ui/widgets/newUserWidget/NewUserWidgetFormContainer.js
Comment (ichalagashvili) [inline src/ui/widgets/newUserWidget/NewUserWidgetFormContainer.js:35]: here to we want `fetchGroups` or `fetchAllGroupEntries` ? or maybe both?

**Comment by entando-jx** [comment]
@jeffgo10: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 32b0b9734e8bebe202d98c32ef6c732dac587196 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@jeffgo10: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | ef83de60eb4c63f08d2d4a33e2cb367492ac241e | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by jeffgo10** [comment]
@ichalagashvili `fetchGroups` has now been renamed to `fetchMyGroups`. Yes you are right, it is better to rename it to acknowledge change around group listing in app builder. But I wanted to ask you about why we need to use the complete group list on widget forms?

**Comment by ichalagashvili** [comment]
hi @jeffgo10 , what is a bit weird for me to understand is that before code was: `dispatch(fetchGroups({ page: 1, pageSize: 0 }));` and you changed it to `dispatch(fetchMyGroups());` .

From my understanding, `fetchGroups` with no pagination was fetching all the available groups to show to user. But now you have replaced them with `fetchMyGroups` which only shows current user's groups. So, the difference is, if we have 10000 groups, initial code would show 10000 groups on the page while your implementation would now show only 1-2 groups user is part of.

**Comment by jeffgo10** [comment]
the provided API endpoint `getMyGroups` does not have any parameters for pagination. According to @BrenoQVDS , the API will just list out all the groups where the user belong.

**Review by ichalagashvili — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/1072#pullrequestreview-667529857" title="Approved">ichalagashvili</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)~~ [ichalagashvili]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
